pandas/core/arrays/base.py

```diff
@@ -49,13 +49,32 @@ class ExtensionArray(object):
     assumptions on how the data are stored, just that it can be converted
     to a NumPy array.
 
-    Extension arrays should be able to be constructed with instances of
-    the class, i.e. ``ExtensionArray(extension_array)`` should return
-    an instance, not error.
+    There are a few restrictions on how ExtensionArrays are created.
+
+    * An ExtensionArray should be valid, i.e.
+      ``ExtensionArray(extension_array)`` should return an instance
+    * A sequence of the scalar type should be valid, i.e.
+      ``ExtensionArray(Sequence[ExtensionDtype.type]])`` should return
+      an instance, not error.
     """
     # '_typ' is for pandas.core.dtypes.generic.ABCExtensionArray.
     # Don't override this.
     _typ = 'extension'
+
+    @classmethod
+    def _constructor(cls, data):
+        """Construct a new instance of of the extension array.
+
+        Parameters
+        ----------
+        data : Sequence
+
+        Returns
+        -------
+        ExtensionArray
+        """
```

Comment:
prob other locations that can be changed to use this

Reply:
The only other `type(self)` in this file is to get the class name.